snapcraft/project/_sanity_checks.py

```diff
@@ -45,6 +47,27 @@ def conduct_project_sanity_check(project: snapcraft.project.Project):
     if os.path.isdir(snap_dir_path):
         _check_snap_dir(snap_dir_path)
 
+    # Unit tests are not seting the base, so we check for that here.
+    if project.info is not None:
+        _migration_notice(project.info.base)
+
+
+def _migration_notice(base: Optional[str]) -> None:
+    # We only want to encourage developers to use bases if using the snapcraft snap.
+    if base is None and common.is_snap():
+        logger.warning(
+            "This snapcraft project does not use bases which means this project "
+            "does not take advantage of the latest snapcraft features.\n"
+            "Read more about bases at https://docs.snapcraft.io/t/base-snaps/11198"
+        )
+    # Or if they are on the deb (this legacy codebase) and using bases.
+    elif base is not None and not common.is_snap():
+        logger.warning(
+            "This version of snapcraft only has experimental support for bases.\n"
+            "Please move to snapcraft 3.0 or greater for proper base support.\n"
```

Comment:
"Proper base support" sounds vague. How about: "Please move to snapcraft 3.0 or greater for the fully supported implementation of bases."